Mutagen.Bethesda.Analyzers.Tests/Config/AnalyzerConfigReaderTests.cs

```diff
@@ -8,33 +10,63 @@ namespace Mutagen.Bethesda.Analyzers.Tests.Config;
 public class AnalyzerConfigReaderTests
 {
     [Theory]
-    [AnalyzerInlineData("diagnostic.A123.severity = Warning")]
-    [AnalyzerInlineData("diagnostic.A123.severity = Warning # A Comment")]
-    public void LinesThatParse(
+    [AnalyzerInlineData("environment.data_directory = C:/some/path")]
+    [AnalyzerInlineData(@"environment.data_directory = C:\some\path")]
+    public void TestDataDirectory(
         string line,
         IAnalyzerConfig config,
         AnalyzerConfigReader sut)
     {
         sut.ReadInto(line.AsSpan(), config);
-        config.Received(1).Override(new TopicId("A", 123), Severity.Warning);
+        config.Received(1).OverrideDataDirectory("C:/some/path");
```

Comment:
If some of the other comments are done.. then this might need to change a bit.   If `AnalyzerConfigReader` is instead the `IConfigReaderProcessorUnit<AnalyzerConfig>`.. then it won't self inject into a reader anymore here.

Could refactor it a few ways, depending on how the API shakes out.. but there's always this trick, if needed:
```cs
public class InternalTestPayload
{
    public readonly IConfigReader<AnalyzerConfig> Reader;

    public InternalTestPayload(AnalyzerConfigProcessingUnit processor)
    {
         // Some common setup
         Reader = new ConfigReader<AnalyzerConfig>(new [] { processor });
    }
}

// ...

[Theory]
    [AnalyzerInlineData("environment.data_directory = C:/some/path")]
    [AnalyzerInlineData(@"environment.data_directory = C:\some\path")]
    public void TestDataDirectory(
        string line,
        AnalyzerConfig config,
        InternalTestPayload payload)
    {
        payload.Reader.ReadInto(line.AsSpan(), config);
        config.DataFolder.Should.Be(...);
```